ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -154,149 +241,511 @@ class Pagination {
+            className: Globals.getClassName('pgWrapper')
         }, this.cell);
 
-        this.renderNavButtons();
+        // Update total pages first to ensure correct calculations
+        this.updateTotalPages();
+
+        // Render all components
+        this.renderPageInfo();
+        this.renderControls();
+        this.renderPageSizeSelector();
+
+        // Update button states after rendering
+        this.updateButtonStates();
     }
 
     /**
-     * Render the navigation buttons.
+     * Render the page information text.
      */
-    public renderNavButtons(): void {
-        const pg = this;
-        const grid = pg.grid;
-        const originalDataTable = grid.dataTable;
+    public renderPageInfo(): void {
+        if (!this.options.controls?.pageInfo) {
+            return;
+        }
 
-        this.prevButton = makeHTMLElement(
-            'button',
-            {
-                innerHTML: '&lsaquo;' // TODO: add lang support
-            },
-            this.contentWrapper
-        );
+        this.pageInfoElement = makeHTMLElement('div', {
+            className: Globals.getClassName('pgPageInfo')
+        }, this.contentWrapper);
 
-        this.setButtonState(
-            this.prevButton,
-            true
-        );
+        this.updatePageInfo();
+    }
 
-        this.prevButton.addEventListener(
-            'click',
-            function (): void {
-                pg.updatePage(false);
-            }
-        );
+    /**
+     * Update the page information text.
+     */
+    public updatePageInfo(): void {
+        if (!this.pageInfoElement) {
+            return;
+        }
 
-        this.nextButton = makeHTMLElement(
-            'button',
-            {
-                innerHTML: '&rsaquo;' // TODO: add lang support
-            },
-            this.contentWrapper
+        const startItem = (this.currentPage - 1) * this.currentPageSize + 1;
+        const endItem = Math.min(
+            this.currentPage * this.currentPageSize,
+            this.totalItems
         );
 
-        this.nextButton.addEventListener(
-            'click',
-            function (): void {
-                pg.updatePage();
-            }
-        );
+        const pageInfoText = this.formatText(this.lang.pageInfo as string, {
+            start: startItem,
+            end: endItem,
+            total: this.totalItems
+        });
+
+        this.pageInfoElement.innerHTML = pageInfoText;
+    }
+
+    /**
+     * Render the controls buttons and page numbers.
+     */
+    public renderControls(): void {
+        const navContainer = makeHTMLElement('div', {
+            className: Globals.getClassName('pgControls')
+        }, this.contentWrapper);
+
+        // Render first/previous buttons
+        if (this.options.controls?.firstLastButtons) {
+            this.renderFirstButton(navContainer);
+        }
+
+        if (this.options.controls?.prevNextButtons) {
```

Comment:
Missing information about what it does (I'm leaving the comment because you did it for all the other renderings).